comp/checks/agentcrashdetect/agentcrashdetect.go

```diff
@@ -0,0 +1,175 @@
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	crashDetectCheckName = "agentcrashdetect"
+	maxStartupWarnings   = 20
+	reportedKey          = `lastReported`
+)
+
+var (
+	// crashdriver included for testing purposes
+	ddDrivers = map[string]struct{}{
+		"ddnpm":       {},
+		"crashdriver": {},
+	}
+	// system probe enabled flags indicating we should be enabled
+	enabledflags = []string{
+		"windows_crash_detection.enabled",
+		"network_config.enabled",
+		"service_monitoring_config.enabled",
+	}
+	// these are vars and not consts so that they can be overridden in
+	// the unit tests.
+	hive    = registry.LOCAL_MACHINE
+	baseKey = `SOFTWARE\Datadog\Datadog Agent\windows_agent_crash_reporting`
+)
+
+// WinCrashConfig is the configuration options for this check
+// it is exported so that the yaml parser can read it.
+type WinCrashConfig struct {
+	Enabled bool `yaml:"enabled"` // placeholder for config
+}
+
+// AgentCrashDetect is the core check object; it implements the core check interface
+// for running agent checks
+type AgentCrashDetect struct {
+	core.CheckBase
+	instance   *WinCrashConfig
+	reporter   *crashreport.WinCrashReporter
+	npmEnabled bool
+	tconfig    *traceconfig.AgentConfig
+}
+
+type agentCrashComponent struct {
+	tconfig *traceconfig.AgentConfig
+}
+
+type dependencies struct {
+	fx.In
+
+	TConfig   comptraceconfig.Component
+	Lifecycle fx.Lifecycle
+}
+
+// Parse parses the check configuration
+func (c *WinCrashConfig) Parse(data []byte) error {
+	// default values
+	c.Enabled = true
+
+	return yaml.Unmarshal(data, c)
+}
+
+// Configure accepts the configuration
+func (wcd *AgentCrashDetect) Configure(senderManager sender.SenderManager, integrationConfigDigest uint64, data integration.Data, initConfig integration.Data, source string) error {
+	err := wcd.CommonConfigure(senderManager, integrationConfigDigest, initConfig, data, source)
+	if err != nil {
+		return err
+	}
+	wcd.reporter, err = crashreport.NewWinCrashReporter(hive, baseKey)
+	if err != nil {
+		return err
+	}
+
+	// check to see if the wincrashdetect module is enabled.  If not, there's no point
+	// in even trying
+	for _, k := range enabledflags {
+		wcd.npmEnabled = config.SystemProbe.GetBool(k)
+		if wcd.npmEnabled {
+			break
+		}
+	}
+	if !wcd.npmEnabled {
+		return fmt.Errorf("Not enabling crash detection module: no relevant configurations enabled")
+	}
+
+	return wcd.instance.Parse(initConfig)
+}
+
+// Run is called on each check run.
+// we're only ever interested in reporting the same crash once.  The reporter.CheckForCrash()
+// will handle only reporting the same crash once, and will return nil, even if a crash
+// is present, if it's already been reported to this check.
+func (wcd *AgentCrashDetect) Run() error {
+
+	crash, err := wcd.reporter.CheckForCrash()
+	if err != nil {
+		return err
+	}
+	if crash == nil {
+		// no crash to send
+		return nil
+	}
+
+	// check to see if the crash is one of ours
+	offender := strings.Split(crash.Offender, "+")[0]
```

Comment:
what does `crash.Offender` look like? can the struct be structured differently so the user doesn't have to do extra string processing?

Reply:
The offender is parsed out of a line like this
`fffff800`45b338f7 : 00000000`00000000 00000000`00000000 ffffc582`1bb72c30 ffffffff`000001c8 : ddapmcrash+0x7020`

In the module that does the parsing, it just takes the 3rd column, so
` ddapmcrash+0x7020`

The crash parsing module is intended to be used by more than one potential consumer, so I don't want to remove the information at that level.  The agent crash detect can decide it only wants the module name rather than module + offset.

Reply:
Could you add 

> fffff80045b338f7 : 0000000000000000 0000000000000000 ffffc5821bb72c30 ffffffff000001c8 : ddapmcrash+0x7020`

and
> ddapmcrash+0x7020

as comments

Reply:
Is more needed than
https://github.com/DataDog/datadog-agent/tree/db/internal-bsod-check/pkg/collector/corechecks/system/wincrashdetect/probe